cvat-ui/src/reducers/settings-reducer.ts

```diff
@@ -394,6 +398,64 @@ export default (state = defaultState, action: AnyAction): SettingsState => {
                 },
             };
         }
+        case SettingsActionTypes.ADD_IMAGE_FILTER: {
+            return {
+                ...state,
+                imageProcessing: {
+                    ...state.imageProcessing,
+                    filters: [
+                        ...state.imageProcessing.filters,
+                        action.payload.filter,
+                    ],
+                },
+            };
+        }
+        case SettingsActionTypes.REMOVE_IMAGE_FILTER: {
+            const { filterAlias } = action.payload;
+            if (filterAlias) {
+                const filters = [...state.imageProcessing.filters];
+                const index = filters.findIndex((imageFilter) => imageFilter.alias === filterAlias);
+                if (index !== -1) {
+                    filters.splice(index, 1);
+                }
+                filters.forEach((imageFilter) => {
+                    imageFilter.modifier.currentProcessedImage = null;
+                });
+                return {
+                    ...state,
+                    imageProcessing: {
+                        ...state.imageProcessing,
+                        filters,
+                    },
+                };
+            }
+            return {
+                ...state,
+                imageProcessing: {
+                    ...state.imageProcessing,
+                    filters: [],
+                },
+            };
+        }
+        case SettingsActionTypes.SETUP_IMAGE_FILTER: {
+            const { filterAlias, options } = action.payload;
+            const filters = [...state.imageProcessing.filters];
+            const index = filters.findIndex((imageFilter) => imageFilter.alias === filterAlias);
+            if (index !== -1) {
+                const filter = filters[index];
+                if (filter.modifier instanceof FabricFilter) {
+                    filter.modifier.currentProcessedImage = null;
+                    filter.modifier.configure(options);
```

Comment:
Let ``configure`` be an interface method of any filter with empty implementation by default.
I believe reducer don't need to decide whether to apply options or not, just apply always.

Reply:
I mean this condition ``if (filter.modifier instanceof FabricFilter)`` is unnecessary here